### PR TITLE
docs: improve model.rst

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- add missing things
- add explanation for "Working with Query Builder"

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by michalsn — APPROVED** [review]

File: user_guide_src/source/models/model.rst [hunk 1]
```diff
@@ -146,49 +151,60 @@ potential mass assignment vulnerabilities.
...
+requires that the table have columns named **created_at**, **updated_at** and **deleted_at** in the appropriate
 data type.
 
+$dateFormat
+^^^^^^^^^^^
+
+This value works with ``$useTimestamps`` and ``$useSoftDeletes`` to ensure that the correct type of
+date value gets inserted into the database. By default, this creates DATETIME values, but
+valid options are: ``'datetime'``, ``'date'``, or ``'int'`` (a PHP timestamp). Using **useSoftDeletes** or
+useTimestamps with an invalid or missing dateFormat will cause an exception.
```
Comment (MGatner) [inline user_guide_src/source/models/model.rst:171]: These concepts should either be all bold or none. 
```suggestion
valid options are: ``'datetime'``, ``'date'``, or ``'int'`` (a PHP timestamp). Using **useSoftDeletes** or
**useTimestamps** with an invalid or missing **dateFormat** will cause an exception.
```

File: user_guide_src/source/models/model.rst [hunk 2]
```diff
@@ -146,49 +151,60 @@ potential mass assignment vulnerabilities.
...
-$dateFormat
------------
+$deletedField
+^^^^^^^^^^^^^
 
-This value works with ``$useTimestamps`` and ``$useSoftDeletes`` to ensure that the correct type of
-date value gets inserted into the database. By default, this creates DATETIME values, but
-valid options are: ``'datetime'``, ``'date'``, or ``'int'`` (a PHP timestamp). Using **useSoftDeletes** or
-useTimestamps with an invalid or missing dateFormat will cause an exception.
+Specifies which database field should use for soft deletions. See :ref:`model-use-soft-deletes`.
```
Comment (MGatner) [inline user_guide_src/source/models/model.rst:188]:
```suggestion
Specifies which database field to use for soft deletions. See :ref:`model-use-soft-deletes`.
```

File: user_guide_src/source/models/model.rst [hunk 3]
```diff
@@ -611,6 +607,57 @@ This is best used during cronjobs, data exports, or other large tasks.
...
+
+In this case, it operates on the shared instance of the Query Builder held by the model.
+
+.. important:: The Model does not provide a perfect interface to the Query Builder.
+    The Model and the Query Builder are separate classes with different purposes.
+    They should not be expected to return the same data.
+
+If the Query Builder returns a result, it is returned as is.
+In that case, the result may be different from the one returned by the model's method
+and may not be what was expected. The model's event is not be triggered.
```
Comment (MGatner) [inline user_guide_src/source/models/model.rst:652]:
```suggestion
and may not be what was expected. The model's events are not triggered.
```

**Review by MGatner — CHANGES_REQUESTED** [review]
Overall very nice improvement. Some suggestions inline.

**Review by MGatner — APPROVED** [review]

**Pull request merged**